composer.json

```diff
@@ -13,7 +13,8 @@
     "require": {
         "php": ">=5.4",
         "php-http/httplug": "^1.0",
-        "php-http/message-factory": "^1.0"
+        "php-http/message-factory": "^1.0",
+        "symfony/options-resolver": "^2.6|^3.0"
```

Comment:
the big benefit of options-resolver is that we are much more flexible to add things without a BC break. adding new constructor parameters can easily lead to BC break, and for these options things, its tricky to decide on the order of them - with separate parameters, the order can not be changed to make logical sense. the array is so much easier.

hm. on the other hand we have only one option for now. maybe we could instead do that manually instead of using OptionsResolver - we can add a requirement for the resolver when we add more options. for just one, its indeed overkill if its not a free include. wdyt? 

* remove the resolver dependency
* keep $options an array
* resolve manually for now
* add a comment next to the options that if we add more options, we should introduce the OptionsResolver